### PR TITLE
Use setup.cfg file instead of setup.py

### PR DESCRIPTION
**Pull request by yannikschaelte**
* No conceptual difference, just kind of "cooler" that the package definition is not in python but pure text.
* Added a bit more meta info (classifiers, URLs, ...)
* I set the minimum python version to 3.7, as e.g. scipy+amici also no longer support 3.6. Opinions? If 3.7 is set as minimum, a few docs need to be updated.
* Also reduced the tests to 3.7+3.9 (min+max). I think that should be better than testing all intermediate versions

TBD: I think the dependencies should be tidied up afterwards, e.g. the test requirement is not used in the actual tests yet.

### CODE REVIEW

**Comment by jvanhoefer** [comment]
Is there any benefit from requiring python > 3.7 or don't we just exclude users for no obvious reason?

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    Programming Language :: Python
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+keywords = YAML, SBML, PEtab, ODE
+
+[options]
+install_requires =
+    numpy >= 1.19.4
+    scipy >= 1.6.0
```
Comment (jvanhoefer) [inline setup.cfg:44]: scipy is actually _not_ used in the public API, but just in the examples...

**Comment by yannikschaelte** [comment]
> Is there any benefit from requiring python > 3.7 or don't we just exclude users for no obvious reason?

I think one can support 3.6, my take was rather "ok, when even the very basic modules like scipy drop 3.6, hardly any user will use 3.6 any more soon, so not worth taking the trouble here". But as some systems still run on 3.6, we can include it here. In that case the tests need to be subdivided though, e.g. if we add tests using AMICI, those need to be restricted to 3.7+.

Most typical applications with PEtab or AMICI or an up-to-date scipy will only work for 3.7+ (and soon 3.8+, which introduces some nice language features). Usually, we only support the last two versions in our packages. If it is only about the conversion YAML -> SBML, that should work with 3.6.

In summary: Both works. Which do you prefer?

File: setup.cfg (hunk 1 above)
Comment (yannikschaelte) [reply setup.cfg:44]: good to know, my mistake! will move it back to the examples.

Comment (yannikschaelte) [reply setup.cfg]: Now it should be exactly as before, also again including python 3.6.

**Review by jvanhoefer — APPROVED** [review]
